applications_superstaq/superstaq_client.py

```diff
@@ -0,0 +1,397 @@
+                from the returned job.
+            target: If supplied the target to run on. Supports one of `qpu` or `simulator`. If not
+                set, uses `default_target`.
+            name: An optional name of the job. Different than the `job_id` of the job.
+
+        Returns:
+            The json body of the response as a dict. This does not contain populated information
+            about the job, but does contain the job id.
+
+        Raises:
+            An SuperstaQException if the request fails.
+        """
+        actual_target = self._target(target)
+        json_dict: Dict[str, Any] = {
+            **serialized_circuits,
+            "backend": actual_target,
+            "shots": repetitions,
+            "ibmq_token": self.ibmq_token,
+            "ibmq_group": self.ibmq_group,
+            "ibmq_project": self.ibmq_project,
+            "ibmq_hub": self.ibmq_hub,
+            "ibmq_pulse": self.ibmq_pulse,
+        }
+
+        def request() -> requests.Response:
+            return requests.post(
+                f"{self.url}/jobs",
+                json=json_dict,
+                headers=self.headers,
+                verify=self.verify_https,
+            )
+
+        return self._make_request(request).json()
+
+    def get_job(self, job_id: str) -> dict:
+        """Get the job from the SuperstaQ API.
+
+        Args:
+            job_id: The UUID of the job (returned when the job was created).
+
+        Returns:
+            The json body of the response as a dict.
+
+        Raises:
+            SuperstaQNotFoundException: If a job with the given job_id does not exist.
+            SuperstaQException: For other API call failures.
+        """
+
+        def request() -> requests.Response:
+            return requests.get(
+                f"{self.url}/job/{job_id}",
+                headers=self.headers,
+                verify=self.verify_https,
+            )
+
+        return self._make_request(request).json()
+
+    def get_balance(self) -> dict:
+        """Get the querying user's account balance in USD.
+
+        Returns:
+            The json body of the response as a dict.
+        """
+
+        def request() -> requests.Response:
+            return requests.get(
+                f"{self.url}/balance",
+                headers=self.headers,
+                verify=self.verify_https,
+            )
+
+        return self._make_request(request).json()
+
+    def get_backends(self) -> dict:
+        """Makes a GET request to SuperstaQ API to get a list of available backends."""
+
+        def request() -> requests.Response:
+            return requests.get(
+                f"{self.url}/backends",
+                headers=self.headers,
+                verify=self.verify_https,
+            )
+
+        return self._make_request(request).json()
+
+    def aqt_compile(self, json_dict: Dict[str, str], target: str) -> dict:
```

Comment:
```suggestion
    def aqt_compile(self, json_dict: Dict[str, Union[str, List[str]]], target: str) -> dict:
```
(because `json_dict["qasm_strs"]` is given as a list of QASM strings)

Reply:
Good. Done.